### DISCUSSION TITLE
Restarting WW3

### DISCUSSION THREAD

**Discussion by benjamin-cash** [NOAA-EMC/WW3#795] [Q&A]
Could someone walk me through or point me to the documentation on how to restart WW3, specifically as part of UFS? I'm currently unable to get UFS Prototype-P8 to restart reproduce and I'm guessing it is because I am missing a step in the WW3 restart process. Thanks!

**Comment by aliabdolali** [comment]
You can specify the restart file intervals under In ww3_shel.inp (https://github.com/NOAA-EMC/WW3/blob/develop/model/inp/ww3_shel.inp#L315) and ww3_multi.inp (https://github.com/NOAA-EMC/WW3/blob/develop/model/inp/ww3_multi.inp#L173).
The model dumps `restartNNN.<grdname>` in the run directory. 
If you want to start another similation with hot start with the restart file(s) from a previous run, you need to copy the corresponding `restartNNN.<grdname>` to the new run directory and rename it to `restart.<grdname>`. The model at initiation step looks for such a file, if exists, it uses it as initial conditon otherwise starts cold.

**Reply by benjamin-cash** [reply]
Thanks @aliabdolali ! One quick question - it appears that ww3_multi.inp is no longer in use, or at least is not in any of the regression tests I have access to. Is that correct, and it has been replaced with ww3_shel.inp?

**Reply by benjamin-cash** [reply]
One more question - could you walk me through the entries in line 321 in ww3_shel.inp so that I know that I am setting them correctly for my own case?

**Comment by aliabdolali** [comment]
You are right, you are using ww3_shel.inp which does the same as ww3_multi.inp.
see the following example:
```
   19680606 030000  3600  19680607 030000 F F
```
**19680606 030000** is the first time step.
**3600** is the time interval between restart files.
**19680607 030000** is the last time step.
the first **F**  is  the Flag for the second restart (data server mode)
the second **F**  Flag for extra fields in the restart file (coupling restart)

Please check your ww3_shel.inp, I think the flags should be T.

**Reply by benjamin-cash** [reply]
Thanks again! What does it mean if the time interval between restart files is set to 0? Does it default to whatever the output frequency is set to? I'm looking at a ww3_shel.inp generated by the cpld_control_p8 regression test and trying to understand how it matches up to the output.

**Reply by aliabdolali** [reply]
0 means no restart. You can either set the first and last time steps outside the simulation period or set interval=0, then the model does noy dump restart.

**Reply by benjamin-cash** [reply]
Hmmm. In the ww3_shel.inp I am looking at, I have the following:

$
$ Type 4 : Restart files (no additional data required).
$                               The data file is restartN.ww3, see
$                               w3iors.ftn for additional doc.
$          Flag for second restart (data server mode)
$          Flag for extra fields in the restart file (coupling restart)
$
$ Keep next two lines formatting as is to allow proper parsing
   20210322 060000   0  21000322 060000 T
20210322 060000   10800  21000322 060000
$
   20210322 060000      0  21000322 060000
$
   20210322 060000      0  21000322 060000
$
  'the_end'  0
$
  'STP'
Orion-login-1[30] cash$ ls -lrt *restart.ww3
-rw-r----- 1 cash nems 178339200 Sep 14 13:31 20210322.060000.restart.ww3
-rw-r----- 1 cash nems 178339200 Sep 14 13:32 20210322.090000.restart.ww3
-rw-r----- 1 cash nems 178339200 Sep 14 13:33 20210322.120000.restart.ww3
-rw-r----- 1 cash nems 178339200 Sep 14 13:33 20210322.150000.restart.ww3
-rw-r----- 1 cash nems 178339200 Sep 14 13:34 20210322.180000.restart.ww3
-rw-r----- 1 cash nems 178339200 Sep 14 13:34 20210322.210000.restart.ww3
-rw-r----- 1 cash nems 178339200 Sep 14 13:35 20210323.000000.restart.ww3
-rw-r----- 1 cash nems 178339200 Sep 14 13:35 20210323.030000.restart.ww3
-rw-r----- 1 cash nems 178339200 Sep 14 13:36 20210323.060000.restart.ww3

**Comment by aliabdolali** [comment]
ok, now I see

We added the second definition for restart file to add the date tag to YYYYMMDD.HHMMSSrestart.ww3 instead of restartNNN.ww3
so in this case, restarts are being written from 20210322 060000, every 3 hour to 21000322 060000.

**Reply by benjamin-cash** [reply]
Ok, so the first line can stay at 0, which is saying no restartNNN.ww3 file, and the 10800 in the second line is controlling the restart frequency for the YYYYMMDD.HHMMSSrestart.ww3 files? Do I have that correct?

**Reply by aliabdolali** [reply]
Correct.

**Reply by benjamin-cash** [reply]
Great, many thanks. :) I will make these changes and see what happens.
